Hazel/FewerThreads/ThreadLimitedUdpServerConnection.cs

```diff
@@ -45,6 +45,10 @@ protected override void WriteBytesToConnection(byte[] bytes, int length)
         {
             if (bytes.Length != length) throw new ArgumentException("I made an assumption here. I hope you see this error.");
 
+            // Hrm, well this is inaccurate for DTLS connections because the Listener does the encryption which may change the size.
+            // but I don't want to have a bunch of client references in the send queue...
+            // Does this perhaps mean the encryption is being done in the wrong class?
```

Comment:
That would work for the total server output, but it would be kinda messy to make the listener report for each connection separately, which was the intended functionality.

Reply:
Ahhh I see, the per-connection Statistics instance and yeah, it might be awkward having the listener manage all that if it is a per-connection stat.

It kind of seems the DLTS layers on the server side are a bit different, it seems to derive just the Listener and manages its connections separately? ie. ThreadLimitedUdpServerConnection isn't subclassed with a corresponding DTLS layer?

Guessing we'll just have to live with the DTLS stats being a bit off for now and think about this more for later

Reply:
Yeah, subclassing the *ServerConnection classes is a bit trickier. I know I ran into some issue when moving from the UdpServerConnection to the ThreadLimited version. Otherwise it seems like such a lean implementation I would have reused it. I imagine that's why there's no DTLS version. There's no need. But if the encryption did belong here, then that would start to make more sense.